Dockerfile

```diff
@@ -16,7 +16,8 @@
 # limitations under the License.
 ################################################################################
 # Build
-FROM maven:3.8.4-eclipse-temurin-11 AS build
+ARG JAVA_VERSION=11
+FROM maven:3.8.8-eclipse-temurin-${JAVA_VERSION} AS build
```

Comment:
Bumps the maven version. I went for the oldest version which supports all three of the desired java versions. Could equally jump to the latest maven version (3.9.9 IIRC). 

This is the core of the fix so that maven in the docker container actually builds with the desired JVM version.